astropy/time/core.py

```diff
@@ -1994,6 +2016,82 @@ class TimeYearDayTime(TimeString):
                 '{year:d}:{yday:03d}'))
 
 
+class TimeFITS(TimeString):
+    """
+    FITS compliant date-time format: "[±Y]YYYY-MM-DDTHH:MM:SS.sss...(SCALE)".
+
+    ISOT with two extensions:
+    - Signed five-digit year (mostly for negative years);
+    - A possible time scale appended in parentheses.
+    """
+    name = 'fits'
+    subfmts = (
+        ('date_hms',
+         '%Y-%m-%dT%H:%M:%S',
+         '{year:04d}-{mon:02d}-{day:02d}T{hour:02d}:{min:02d}:{sec:02d}'),
+        ('date_hm',
+         '%Y-%m-%dT%H:%M',
+         '{year:04d}-{mon:02d}-{day:02d}T{hour:02d}:{min:02d}'),
+        ('date',
+         '%Y-%m-%d',
+         '{year:04d}-{mon:02d}-{day:02d}'),
+        ('longdate_hms',
+         r'(?P<year>[+-]\d{1,5})-%m-%dT%H:%M:%S',
```

Comment:
My reading of the spec is that dates like `+12-01-01` are not allowed.  Furthermore I think that one-digit versions of the month, hour etc are not allowed, which would imply writing the whole thing as a regex instead of using the more permissive `strptime` specifiers.

All of the time part may be omitted (just leaving the date) or the decimal seconds may be omitted. Leading zeroes must not be omitted and timezone designators are not allowed.
This paper extends the definition to allow five-digit years with a (mandatory) sign, in accordance with ISO-8601. I.e., one shall either use the unsigned four-digit year format or the signed five-digit year format:
[±C]CCYY-MM-DD[Thh:mm:ss[.s...]]